code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -448,14 +448,11 @@
 	broken = TRUE //applies to secure lockers only
 	open()
 
-/obj/structure/closet/AltClick(mob/user)
-	..()
-	if(!user.canUseTopic(src, BE_CLOSE) || !isturf(loc))
-		return
+/obj/structure/closet/attack_hand_secondary(mob/user, modifiers)
 	if(opened || !secure)
-		return
-	else
-		togglelock(user)
+		return SECONDARY_ATTACK_CONTINUE_CHAIN
```

Comment:
I don't think it makes much sense to continue `afterattack` when nothing has happened. This entire thing should probably just check if it *can* lock, lock it, then always return `SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN`.